llvm/lib/Transforms/Utils/SimplifyCFG.cpp

```diff
@@ -1608,16 +1622,42 @@ bool SimplifyCFGOpt::hoistCommonCodeFromSuccessors(BasicBlock *BB,
 
   auto *TI = BB->getTerminator();
 
+  SmallVector<BasicBlock *, 8> SuccessorBBs;
+  for (auto *Succ : successors(BB)) {
+    // If we find an unreachable instruction at the beginning of a basic block,
+    // we can still hoist instructions from the rest of the basic blocks.
+    if (isa<UnreachableInst>(Succ->getFirstNonPHIOrDbg()))
+      continue;
+    SuccessorBBs.push_back(Succ);
+  }
+
+  // Find the smallest BB because we always want to iterate over instructions
+  // of the smallest Successor.
+  auto *SmallestBBPos =
+      std::min_element(SuccessorBBs.begin(), SuccessorBBs.end(),
+                       [](BasicBlock *BB1, BasicBlock *BB2) {
+                         return BB1->size() < BB2->size();
+                       });
+
+  std::iter_swap(SuccessorBBs.begin(), SmallestBBPos);
+  BasicBlock *SmallestBB = *SmallestBBPos;
+
+  if (SmallestBB->size() > BBSizeHoistLimit)
```

Comment:
These size() calls should be sizeWithoutDebug(), otherwise the debug invariance property will be violated.